Tools/clinic/clinic.py

```diff
@@ -828,6 +828,31 @@ class CLanguage(Language):
             #define {methoddef_name}
         #endif /* !defined({methoddef_name}) */
     """)
+    DEPRECATED_POSITIONAL_PROTOTYPE: Final[str] = r"""
+        #if PY_MAJOR_VERSION == {major} && \
+            PY_MINOR_VERSION == {minor} && \
+            PY_RELEASE_LEVEL == PY_RELEASE_LEVEL_ALPHA
+        #  ifdef _MSC_VER
+        #    pragma message ("{cpp_warning}")
+        #  else
+        #    warning "{cpp_warning}"
+        #  endif
+        #elif PY_MAJOR_VERSION > {major} || \
+             (PY_MAJOR_VERSION == {major} && PY_MINOR_VERSION > {minor}) || \
+             (PY_MAJOR_VERSION == {major} && \
+              PY_MINOR_VERSION == {minor} && \
+              PY_RELEASE_LEVEL == PY_RELEASE_LEVEL_GAMMA)
+        #  error "{cpp_warning}"
+        #endif
+        if (nargs == {pos}) {{{{
+            if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                "Using {name!r} as a positional argument is deprecated. "
+                "It will become a keyword-only argument in Python {thenceforth}.", 2))
```

Comment:
Keyword-only *parameter*, not *argument*.

```suggestion
                "It will become a keyword-only parameter in Python {thenceforth}.", 2))
```

Reply:
Note that PEP-3102 uses _argument_ both for positional and keyword parameters.